frontend/context/StakingProgramProvider.tsx

```diff
@@ -1,73 +1,88 @@
-/*
-  This context provider is responsible for determining the current active staking program, if any.
-  It does so by checking if the current service is staked, and if so, which staking program it is staked in.
-  It also provides a method to update the active staking program id in state.
-*/
+/**
+ * This context provider is responsible for determining the current active staking program, if any.
+ * It does so by checking if the current service is staked, and if so, which staking program it is staked in.
+ * It also provides a method to update the active staking program id in state.
+ */
 
-import { createContext, PropsWithChildren, useCallback, useState } from 'react';
-import { useInterval } from 'usehooks-ts';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { createContext, PropsWithChildren, useCallback } from 'react';
 
-import { CHAIN_CONFIG } from '@/config/chains';
+import { AGENT_CONFIG } from '@/config/agents';
+import { GNOSIS_CHAIN_CONFIG } from '@/config/chains';
+import { FIVE_SECONDS_INTERVAL } from '@/constants/intervals';
+import { REACT_QUERY_KEYS } from '@/constants/react-query-keys';
 import { StakingProgramId } from '@/enums/StakingProgram';
+import { useService } from '@/hooks/useService';
 import { useServices } from '@/hooks/useServices';
-import { AutonolasService } from '@/service/Autonolas';
 
-export const INITIAL_DEFAULT_STAKING_PROGRAM_ID = StakingProgramId.Beta;
+export const INITIAL_DEFAULT_STAKING_PROGRAM_ID = StakingProgramId.PearlBeta;
 
 export const StakingProgramContext = createContext<{
   activeStakingProgramId?: StakingProgramId | null;
   defaultStakingProgramId: StakingProgramId;
-  updateActiveStakingProgramId: () => Promise<void>;
-  setDefaultStakingProgramId: (stakingProgramId: StakingProgramId) => void;
 }>({
   activeStakingProgramId: undefined,
   defaultStakingProgramId: INITIAL_DEFAULT_STAKING_PROGRAM_ID,
-  updateActiveStakingProgramId: async () => {},
-  setDefaultStakingProgramId: () => {},
 });
 
-/** Determines the current active staking program, if any */
-export const StakingProgramProvider = ({ children }: PropsWithChildren) => {
-  const { service } = useServices();
+const currentAgent = AGENT_CONFIG.trader; // TODO: replace with dynamic agent selection
+const currentChainId = GNOSIS_CHAIN_CONFIG.chainId; // TODO: replace with dynamic chain selection
 
-  const [activeStakingProgramId, setActiveStakingProgramId] =
-    useState<StakingProgramId | null>();
+/**
+ * hook to get the active staking program id
+ */
+const useGetActiveStakingProgramId = () => {
+  const queryClient = useQueryClient();
 
-  const [defaultStakingProgramId, setDefaultStakingProgramId] = useState(
-    INITIAL_DEFAULT_STAKING_PROGRAM_ID,
-  );
```

Comment:
this value is really important for users that're onboarding, as they are not actively staked

Reply:
the middleware returns an empty array when there're no services

without a service we cannot use activeStakingProgramId, as there is no actively staked services, hence the placeholder of `defaultStakingProgramId` is used as a fallback in places requiring a staking program id, i.e. deploy agent button, initial funding, migration page, .etc